### ISSUE TITLE
Not creating `yarn.lock` on `yarn install`

### ISSUE THREAD

**Issue by pedro-mass** [yarnpkg/yarn#5716]
**Do you want to request a *feature* or report a *bug*?**	bug

**What is the current behavior?**
`yarn install` isn't producing a `yarn.lock`, thus preventing `yarn upgrade` 

**If the current behavior is a bug, please provide the steps to reproduce.**

- `npx create-react-app my-app`
- `cd my-app`
- `yarn install`
- `yarn upgrade`

![image](https://user-images.githubusercontent.com/1165961/39128687-71938bd6-46d6-11e8-90e6-3624418f2120.png)

**What is the expected behavior?**
`yarn.lock` created when running `yarn install`

**Please mention your node.js, yarn and operating system version.**
node: v8.11.1
yarn: 1.6.0
OS: macOS Sierra v10.12.6

**Comment by amphro** [comment]
I ran into this as well. Right after running `yarn install`, I don't have a yarn.lock file and I get errors when running commands that need it. 

```
-> yarn why chalk
yarn why v1.5.1
error An unexpected error occurred: "No lockfile in this directory. Run `yarn install` to generate one.".
info If you think this is a bug, please open a bug report with the information provided in ".../yarn-error.log".
info Visit https://yarnpkg.com/en/docs/cli/why for documentation about this command.
```

**Comment by jack-guy** [comment]
For those who are still stuck on this, running `yarn add` (for any package) _will_ add the lockfile.

**Comment by dovidweisz** [comment]
Another option is to create an empty lock-file before running yarn install.

**Comment by TroyWolf** [comment]
I tried various suggestions with no luck. Then I did this:

`rm -rf node_modules`  <-- blows away your entire node_modules directory
`yarn install`

A nice, fresh yarn.lock file was the result.

**Comment by paul-soporan** [comment]
Closing as fixed in v2.

https://yarnpkg.com/getting-started/migration

**Comment by paulrberg** [comment]
I tried both of the following:

1. `rm -rf node_modules && yarn install`
2. `touch yarn.lock && yarn install`

None worked. Using yarn@1.22.10 on macOS@10.15.7.

**Comment by paulrberg** [comment]
Buuut after I migrated to yarn v2 and kept node_modules via the following setting in `.yarnrc.yml`:

```yml
nodeLinker: node-modules
```

It worked.

**Comment by Xetera** [comment]
If you're here from google, it's possible that you're using yarn workspaces which will hoist the lockfile to the workspace root and make it seem like yarn is not creating a lockfile when you install.

**Comment by tarikhamilton** [comment]
> If you're here from google, it's possible that you're using yarn workspaces which will hoist the lockfile to the workspace root and make it seem like yarn is not creating a lockfile when you install.

This was my issue, thanks! I still had problems after I removed my workspace's `yarn.lock` file and cleared my yarn cache and everything. To resolve this without removing the package from the workspace's packages list, I added `yarn.lock` to the `nohoist` property in my yarn workspace package.json. I'm not sure if there any undesired side effects.

**Comment by SaitoBP** [comment]
I'm using yarn with lerna, and no yarn.lock is created even in the workspace root. Any sugestions? I have already uninstalled yarn and purged all the node_modules folders from the workspace

**Comment by tzipifreedman** [comment]
@SaitoBP I'm facing exactly the same issue - 
I'v just migrate to yarn v3.2.1 deleted the yarn.lock and node_modules then after running `yarn install` none of them was recreated.

**Comment by radoslawszymula** [comment]
> If you're here from google, it's possible that you're using yarn workspaces which will hoist the lockfile to the workspace root and make it seem like yarn is not creating a lockfile when you install.

Ok and how do you fix this?

**Comment by paulcoffee85** [comment]
> I tried various suggestions with no luck. Then I did this:
> 
> `rm -rf node_modules` <-- blows away your entire node_modules directory `yarn install`
> 
> A nice, fresh yarn.lock file was the result.

You are the wisest yarn man.

**Comment by monomelodies** [comment]
For anyone (like me ;-)) running into this, scratching their heads and not getting it work as expected:

My package.json referred to a package in a version that was incompatible with the version of Node I am running (no, I currently cannot upgrade, still stuck on v14 unfortunately, but that's beside the point). Whilst this _seemed_ like an innocent (and barely noticeable) warning, it actually caused the _entire_ operation to revert. Thus, no lockfile, no updated node_modules, nothing.

This also explained a whole lot of other mysterious errors I was running into. Being stuck on an unexpected older version of a package (since nothing was at the version advertised in the lockfile anymore)... doesn't help.

I force-installed a know-to-work version of the offending package, and the error disappeared (and the lockfile reappeared).

Tl;dr: on _any_ error, Yarn will rollback _everything_, not keep the bits that _did_ succeed.

**Comment by ani-hovhannisyan** [comment]
For my case similar to some cases here, nothing worked until I fixed installation errors (Node version incompatibility).
The "yarn.lock" file wasn't created because I had error during installation, as "yarn install" wasn't successfully completed it does not create any file.
Once I fixed the errors (updated Node v10 to Node v14), "yarn install" created yarn.lock file automatically.

**Comment by wbsdickson** [comment]
> I tried both of the following:
> 
> 1. `rm -rf node_modules && yarn install`
> 2. `touch yarn.lock && yarn install`
> 
> None worked. Using yarn@1.22.10 on macOS@10.15.7.

tried, no luck still

**Comment by mczuchryta** [comment]
@wbsdickson check if you have `--install.frozen-lockfile true` in `.yarnrc` that was my case

**Comment by wbsdickson** [comment]
> @wbsdickson check if you have `--install.frozen-lockfile true` in `.yarnrc` that was my case

Finally I solved it by restarting webstorm coz I found that it works properly in a standalone windows terminal.
